Data/Primitive/Array.hs

```diff
@@ -762,7 +762,7 @@ instance Semigroup (Array a) where
 instance Monoid (Array a) where
   mempty = empty
 #if !(MIN_VERSION_base(4,11,0))
-  mappend = (<|>)
```

Comment:
Shouldn't the version guard be sufficient?

Reply:
This change (to `(<>)` which is defined as `(<|>)`) is to silence the `noncanonical-mappend` warning under GHC 8.0 and 8.2 .

Reply:
Well, if you ask GHC 8.0.2, it is not happy about `mappend = (<|>)`.  
```shell-session
$ cabal build -w ghc-8.0.2
...
Data/Primitive/Array.hs:765:3: warning: [-Wnoncanonical-monoid-instances]
    Noncanonical ‘mappend’ definition detected
    in the instance declaration for ‘Monoid (Array a)’.
    Define as ‘mappend = (<>)’
```
If I lower the version in the CPP to `base-4.9.0` (which removes the `mappend` definition), GHC 8.0.2 isn't happy either:
```shell-session
$ cabal build -w ghc-8.0.2
...
Data/Primitive/Array.hs:762:10: warning: [-Wmissing-methods]
    • No explicit implementation for
        ‘mappend’
    • In the instance declaration for ‘Monoid (Array a)’
```
GHC 8.0.2 is like a defiant child, it wants its way, which is `mappend = (<>)`. Basta!

Reply:
Wow, that was lousy behavior indeed!